src/borrow_tracker/tree_borrows/diagnostics.rs

```diff
@@ -80,7 +80,7 @@ pub struct Event {
 /// Diagnostics data about the current access and the location we are accessing.
 /// Used to create history events and errors.
 #[derive(Clone, Debug)]
-pub struct DiagnosticsExtra {
+pub struct AccessDiagnostics {
```

Comment:
I'm not a big fan of this name -- "diagnostic" is the term used in the Rust compiler for a fully computed warning/error output, and that's not what this is.

What about "DiagnosticInfo"?